pkg/localclusterimport/import_local_cluster.go

```diff
@@ -0,0 +1,262 @@
+package localclusterimport
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	hiveext "github.com/openshift/assisted-service/api/hiveextension/v1beta1"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/openshift/hive/apis/hive/v1/agent"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Local_cluster_import struct {
+	clusterImportOperations ClusterImportOperations
+	localClusterNamespaceName    string
+	log                          *logrus.Logger
+}
+
+func NewImportLocalCluster(localClusterImportOperations ClusterImportOperations, log *logrus.Logger, localClusterNamespaceName string) Local_cluster_import {
+	return Local_cluster_import{clusterImportOperations: localClusterImportOperations, log: log, localClusterNamespaceName: localClusterNamespaceName}
+}
+
+func (i *Local_cluster_import) shouldImportLocalCluster() (bool, error) {
+	// The presence of agentClusterInstall for the local cluster will determine whether or not we need to perform an import
+	agentClusterInstall, err := i.clusterImportOperations.GetAgentClusterInstall(i.localClusterNamespaceName, i.localClusterNamespaceName+"-cluster-install")
+	if err != nil && !k8serrors.IsNotFound(err) {
+		i.log.Errorf("There was an error fetching the AgentClusterInstall while checking for need to register local cluster.")
+		return false, err
+	}
+	if agentClusterInstall != nil {
+		i.log.Infof("Found AgentClusterInstall for hub cluster, assuming that hub has been correctly registered for ZTP day 2 operations: %s", agentClusterInstall.Name)
+		return false, nil
+	}
+	return true, nil
+}
+
+func (i *Local_cluster_import) ImportLocalCluster() error {
+	shouldImportLocalCluster, err := i.shouldImportLocalCluster()
+	if err != nil {
+		return err
+	}
+	if !shouldImportLocalCluster {
+		message := "no need to import local cluster as registration already detected"
+		i.log.Info(message)
+		return errors.New(message)
+	}
+	namespace, err := i.clusterImportOperations.GetNamespace(i.localClusterNamespaceName)
+	if err != nil {
+		message := fmt.Sprintf("hubClusterDay2Importer: Unable to find the %s namesapce due to error: %s", i.localClusterNamespaceName, err.Error())
+		i.log.Error(message)
+		return err
+	}
+	if namespace != nil {
```

Comment:
in general for get operations is there a case where the error was nil and the object is nil as well?

Reply:
good point, I will remove the unnecessary checks